### PR TITLE
Changes for substrate#3699

### PR DESCRIPTION
**Pull request by expenses**
See https://github.com/paritytech/substrate/pull/3699.

### CODE REVIEW

**Comment by bkchr** [comment]
> should still point to `polkadot-master`, once we've backported there.

Yeah, but that can only be done by team members. The current approach is okay, but the tests are still failing. We can update `polkadot-master`, when the CI is happy here and replace the branch again.

**Comment by expenses** [comment]
Anything I should be doing to fix the failing tests?

**Comment by rphmeier** [comment]
The test is dependent on the random seed, I suppose. What could have changed it here?

**Comment by bkchr** [comment]
Yeah, I would assume the same. So, you would need to look into this please :)

**Comment by expenses** [comment]
Done :^)

File: runtime/src/parachains.rs [hunk 1]
```diff
@@ -1540,16 +1541,19 @@ mod tests {
...
 
 
 			System::initialize(&2, &H256::from([2; 32]), &Default::default(), &Default::default());
+			RandomnessCollectiveFlip::on_initialize(1);
 			let duty_roster_2 = Parachains::calculate_duty_roster().0;
 			check_roster(&duty_roster_2);
-			assert!(duty_roster_0 != duty_roster_2);
-			assert!(duty_roster_1 != duty_roster_2);
+			assert_ne!(duty_roster_0, duty_roster_2);
+			assert_ne!(duty_roster_1, duty_roster_2);
```
Comment (rphmeier) [inline runtime/src/parachains.rs]: woah, that doesn't help.

**Comment by bkchr** [comment]
@expenses I updated the `polkadot-master` branch in Substrate to your commit. Could you please reset everything back to use this branch? Then we should be good to go.

**Pull request merged**